### PR TITLE
Clean up submit_IM_calc and add ims to the root defaults template

### PR DESCRIPTION
**Pull request by jpa198**
Breaks compatibility with old installs
Draft with the current direction I'm going with it
#### Checklist
- Have you updated the README (yes/no)?
- Have you updated the CHANGELOG (yes/no)?

### CODE REVIEW

File: scripts/submit_sim_imcalc.py [hunk 1]
```diff
@@ -89,237 +36,146 @@ def submit_im_calc_slurm(
...
+            command_options[const.SlBodyOptConsts.simple_out.value] = "-s"
 
-    submit_yes = (
-        True if options_dict["auto"] else confirm("Also submit the job for you?")
-    )
+        if params["ims"][const.RootParams.extended_period.name]:
+            command_options[const.SlBodyOptConsts.extended.value] = "-e"
+            period_count = 100
+        else:
+            period_count = 15
```
Comment (jasonmotha) [inline scripts/submit_sim_imcalc.py:122]: 31 / 131 But yeah not generally a fan of this. Unsure how to do it better though.

**Review by jasonmotha — COMMENTED** [review]
*thumbsup*

File: scripts/submit_sim_imcalc.py (hunk 1 above)
Comment (jpa198) [reply scripts/submit_sim_imcalc.py:122]: Oh, pSA_periods is a variable I can take the length of. Getting the EE period count as a variable would probably require importing from IM_calc though

Comment (jasonmotha) [reply scripts/submit_sim_imcalc.py:122]: That sounds promising. Don't you also need to import from IM_calc at some point anyway? Or is it separated well enough?

Comment (jpa198) [reply scripts/submit_sim_imcalc.py]: Tada, just needed an import from qcore.constants

**Comment by jasonmotha** [comment]
Wouldn't we be better off upgrading the Jenkins version of black?

**Comment by jpa198** [comment]
> Wouldn't we be better off upgrading the Jenkins version of black?

Probably, we should get Sung to do it when there aren't many PRs going, then give each branch an 'update black' PR
It's a pip package, so there shouldn't be any platform issues

**Comment by jpa198** [comment]
@jasonmotha  Nope, looks like it's fine, and my local black was just being weird. It's all fine now, and no upgrades needed anywhere

**Comment by jpa198** [comment]
Turns out this is the whole PR, my original refactor did most of the leg work

**Comment by jpa198** [comment]
Qcore PR must go in first

**Review by claudio525 — APPROVED** [review]

File: scripts/cybershake/auto_submit.py [hunk 2]
```diff
@@ -56,6 +55,8 @@ def submit_task(
     if not os.path.isdir(ch_log_dir):
         os.mkdir(ch_log_dir)
 
+    params = utils.load_sim_params(os.path.join(sim_dir, "sim_params.yaml"))
```
Comment (shot0829) [inline scripts/cybershake/auto_submit.py:58]: although its unlikely to change, but wouldn't it be better if we use the defined structure from qcore.simulation_structure?

Comment (jpa198) [reply scripts/cybershake/auto_submit.py]: Good point, changed

File: scripts/obs_im_calc.sl [hunk 3]
```diff
@@ -23,7 +23,7 @@ do
     if [[  `find ${D} -name accBB | wc -l` -ge 1 ]]
     then
         fault_name=`basename $D`
-        python $gmsim/workflow/scripts/im_calc_checkpoint.py ${obs_dirs}/IM_calc/ $((`ls $D/*/*/accBB | wc -l` / 3)) --event_name ${fault_name} --observed
+        python $gmsim/workflow/scripts/im_calc_checkpoint.py ${obs_dirs}/IM_calc/ $((`ls $D/*/*/accBB | wc -l` / 3)) 1 --event_name ${fault_name} --observed
```
Comment (shot0829) [inline scripts/obs_im_calc.sl:26]: to make it easier for future usage without looking into the checkpoint script what '1' stands for,
i suggest making the comp a variable.
for example
```suggestion
       comp="000 090 geom"
       comp_count=`echo $comp | awk '{print NF}' `
       python $gmsim/workflow/scripts/im_calc_checkpoint.py ${obs_dirs}/IM_calc/ $((`ls $D/*/*/accBB | wc -l` / 3)) $comp_count --event_name ${fault_name} --observed
```
and 
```suggestion
        time python ${IMPATH}/calculate_ims.py $D/*/*/accBB a -o ${obs_dirs}/IM_calc/ -np ${SLURM_CPUS_PER_TASK} -i ${fault_name} -r ${fault_name} -c $comp -t o -e -s
```
default should still be geom BTW, just using 3 comps to show how it works.

**Review by shot0829 — APPROVED** [review]

**Review by claudio525 — APPROVED** [review]

**Pull request merged**